pkg/clusteragent/clusterchecks/dispatcher_nodes.go

```diff
@@ -43,3 +44,50 @@ func (d *dispatcher) processNodeStatus(nodeName string, status types.NodeStatus)
 
 	return (node.lastConfigChange == status.LastChange), nil
 }
+
+// getLeastBusyNode returns the name of the node that is assigned
+// the lowest number of checks. In case of equality, one is chosen
+// randomly, based on map iterations being randomized.
+func (d *dispatcher) getLeastBusyNode() string {
+	var leastBusyNode string
+	minCheckCount := int(-1)
+
+	d.store.RLock()
+	defer d.store.RUnlock()
+
+	for name, store := range d.store.nodes {
+		if name == "" {
+			continue
+		}
+		if minCheckCount == -1 || len(store.digestToConfig) < minCheckCount {
+			leastBusyNode = name
+			minCheckCount = len(store.digestToConfig)
+		}
+	}
+	return leastBusyNode
+}
+
+// expireNodes iterates over nodes and removes the ones that have not
+// reported for more than the expiration duration. The configurations
+// dispatched to these nodes will be moved to the danglingConfigs map.
+func (d *dispatcher) expireNodes() {
+	cutoffTimestamp := timestampNow() - d.nodeExpirationSeconds
+
+	d.store.Lock()
+	defer d.store.Unlock()
+
+	for name, node := range d.store.nodes {
+		node.RLock()
+		if node.lastPing < cutoffTimestamp {
```

Comment:
I didn't catch that earlier, but can we talk briefly about the `lastPing`?
If I understand correctly we use the ping from the client (that checks connection with the cluster agent, and triggers the polling of new configs) as a heartbeat. Is that correct? If so we should explain that somewhere, and maybe rename lastPing in the DCA code to heartbeat. lastPing is fine client side but heartbeat would work as well, your call.
Let me know if I missed something.